src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs

```diff
@@ -77,34 +77,7 @@ public SafeHandle Handle
         /// </remarks>
         public static ThreadPoolBoundHandle BindHandle(SafeHandle handle)
         {
-            if (handle == null)
-                throw new ArgumentNullException(nameof(handle));
-
-            if (handle.IsClosed || handle.IsInvalid)
-                throw new ArgumentException(SR.Argument_InvalidHandle, nameof(handle));
-
-            try
-            {
-                // ThreadPool.BindHandle will always return true, otherwise, it throws. See the underlying FCall
-                // implementation in ThreadPoolNative::CorBindIoCompletionCallback to see the implementation.
-                bool succeeded = ThreadPool.BindHandle(handle);
```

Comment:
I'm fine with making ThreadPool.BindHandle throw PNSE though if that's what we want